sputnikdao2/src/delegation.rs

```diff
@@ -31,12 +33,19 @@ impl Contract {
             .delegations
             .get(account_id)
             .expect("ERR_NOT_REGISTERED");
-        self.delegations
-            .insert(account_id, &(prev_amount + amount.0));
+        let new_amount = prev_amount + amount.0;
+        self.delegations.insert(account_id, &new_amount);
         self.total_delegation_amount += amount.0;
+        (
+            U128(prev_amount),
+            U128(new_amount),
+            U128(self.total_delegation_amount),
```

Comment:
Can you use `self.delegation_total_supply()` instead of `U128(self.total_delegation_amount)`?